src/models/image-map.ts

```diff
@@ -417,16 +446,69 @@ export const localAssetsImagesHandler: IImageHandler = {
     return url ? url.startsWith("assets/") || url.startsWith("curriculum/") : false;
   },
 
-  async store(url: string) {
+  async store(url: string, curriculumBaseUrl?: string, unitCodeMap?: any, options?: IImageHandlerStoreOptions) {
+    const unitCode = urlParams.unit && isValidHttpUrl(urlParams.unit)
+                     ? getUnitCodeFromUrl(urlParams.unit)
+                     : urlParams.unit
+                       ? urlParams.unit
+                       : "sas";
+    const curriculumBranch = urlParams.unit && isValidHttpUrl(urlParams.unit)
+                               ? getCurriculumBranchFromUrl(urlParams.unit)
+                               : undefined;
+    // Legacy curriculum units lived in the defunct "curriculum" directory in subfolders
+    // named after the unit title, e.g. "curriculum/stretching-and-shrinking". References
+    // to files in those directories will need to be converted to use the new file structure
+    // which uses the unit code as the directory name, e.g. "sas".
+    const urlPieces = url.split("curriculum/");
+    const urlUnitDir = urlPieces[1] ? urlPieces[1].replace(/\/.*/, "") : "";
```

Comment:
I'd suggest using a pattern like: `const urlPieces = url.match(/curriculum\/([^\/]+)\/(.*)/)` then you can pull out the first match group which would be the `urlUnitDir`.
And the second match group is the rest of the url.
So now you could assign _url to be `${newUnitDir}/urlPieces[2]`. 

You'd still have handle the `"assets/tools/drawing-tool/stamps"` specially since it wouldn't match the pattern.